apm-agent-core/src/main/java/co/elastic/apm/agent/impl/baggage/RecyclableCharArray.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.impl.baggage;
+
```

Comment:
Why is this class in this package?

Reply:
It was added as a replacement for the `ThreadLocal` usage within `PercentEncoder`. However, after this PR is merged I'll rebase #3239 and replace this class with the easier pooling added in that PR.

Reply:
yes, I mean why THIS package rather than the one with PercentEncoder?